code/tools/signalinjector/src/signalinjector.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (C) 2016 Chalmers REVERE
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+#include <algorithm>  
+#include <ctype.h>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <vector>
+#include <unistd.h>
+
+#include "opendavinci/GeneratedHeaders_OpenDaVINCI.h"
+#include "opendavinci/odcore/data/Container.h"
+#include "opendavinci/odcore/data/TimeStamp.h"
+#include "opendavinci/odcore/base/KeyValueConfiguration.h"
+#include "opendavinci/odcore/strings/StringToolbox.h" 
+
+#include "opendlvdata/GeneratedHeaders_opendlvdata.h"
+
+#include "signalinjector.hpp"
+
+
+
+namespace opendlv {
+namespace tools {
+namespace signalinjector {
+
+
+Signalinjector::Signalinjector(int32_t const &a_argc, char **a_argv)
+    : odcore::base::module::TimeTriggeredConferenceClientModule(
+      a_argc, a_argv, "tools-signalinjector"),
+      m_initialized(false),
+      m_fileNames(),
+      m_testNumber(),
+      m_brake(),
+      m_steering(),
+      m_throttle()
+{
+}
+
+Signalinjector::~Signalinjector()
+{
+}
+
+void Signalinjector::setUp()
+{
+  odcore::base::KeyValueConfiguration kv = getKeyValueConfiguration();
+  std::string const filenames = kv.getValue<std::string>("tools-signalinjector.filenames");
+  m_fileNames = odcore::strings::StringToolbox::split(filenames,',');
+  // std::cout << m_fileNames[0] << std::endl;
+  m_testNumber = kv.getValue<int32_t>("tools-signalinjector.testnumber"); 
+  // std::cout << m_testNumber << std::endl;
+
+  std::string path = std::to_string(m_testNumber) + "/";
```

Comment:
I think that we should make this new feature a bit more general before putting it to master. For example I would like it to handle any message from the odvd. I'm currently doing something similar in what I call a SignalAdapter, which is basically the same thing but uses read and write via UDP instead of CSV. The injector should of course be write only, but I think it should handle any signal.